editor/editor_node.cpp

```diff
@@ -606,8 +606,10 @@ void EditorNode::_notification(int p_what) {
 				_editor_select(EDITOR_3D);
 			}
 
-			// Save the project after opening to mark it as last modified.
-			ProjectSettings::get_singleton()->save();
+			// Save the project after opening to mark it as last modified, except in headless mode.
```

Comment:
It seems a bit weird that we're saving the project just to mark it as last modified in the project manager. I would prefer not saving the project for this reason, even if it means changing the project manager behavior to actually show the last *modified* project first instead of the last *opened*.

Reply:
Yeah while I understand where user requests are coming from, IMO it was a pretty bad hack to save the project without any modification just for this purpose - and as we see it's not without drawbacks.

If we want a "last opened" sorting, this should be achieved through metadata in the editor settings dir, not by tweaking "last modified" to cover both.

Reply:
Well, we could store the opening time somewhere in the editor data, but IMO it's not really such a big problem

Reply:
I'm in favor of removing this and just using *last modified*.

Reply:
IIRC the last time we removed the "save on open" behavior it resulted in a lot of things not working as expected, along with the obvious of the project not being marked as modified. I would prefer leaving this functionality in personally.

Reply:
Yep, there is a bunch of stuff that does some changes at startup (like removing invalid plugins etc.) and it doesn't save the project, it relies on the autosaving. So if we were to remove that, we have to make sure that all these startup modifications are saved.